internal/impl/kv/processor_kv.go

```diff
@@ -0,0 +1,103 @@
+package kv
+
+import (
+	"context"
+	"github.com/benthosdev/benthos/v4/public/service"
+	"regexp"
+)
+
+func init() {
+	if err := service.RegisterProcessor(
+		"kv",
+		newKVProcessorConfig(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.Processor, error) {
+			return newKVProcessor(conf, mgr)
+		},
+	); err != nil {
+		panic(err)
+	}
+}
+
+type kvProcessor struct {
+	logger *service.Logger
+
+	pairDelimiter     string
+	keyValueSeparator string
+	targetField       string
+}
+
+func newKVProcessor(conf *service.ParsedConfig, mgr *service.Resources) (*kvProcessor, error) {
+	logger := mgr.Logger()
+
+	pairDelimiter, err := conf.FieldString("pair_delimiter")
+	if err != nil {
+		return nil, err
+	}
+
+	keyValueSeparator, err := conf.FieldString("key_value_separator")
+	if err != nil {
+		return nil, err
+	}
+
+	var targetField string
+	if conf.Contains("target_field") {
+		targetField, err = conf.FieldString("target_field")
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		targetField = ""
+	}
+
+	return &kvProcessor{
+		logger:            logger,
+		pairDelimiter:     pairDelimiter,
+		keyValueSeparator: keyValueSeparator,
+		targetField:       targetField,
+	}, nil
+}
+
+func newKVProcessorConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Version("4.16.0").
```

Comment:
This should be changed to 4.25 which is the next release

Reply:
updated